pkg/importer/hcl2_test.go

```diff
@@ -263,22 +263,34 @@ func TestGenerateHCL2Definition(t *testing.T) {
 
 			snapshot := []*resource.State{
 				{
-					ID:     "123",
-					Custom: true,
-					Type:   "pulumi:providers:aws",
-					URN:    "urn:pulumi:stack::project::pulumi:providers:aws::default_123",
+					ID:             "123",
+					ImportID:       "abc",
+					Custom:         true,
+					Type:           "pulumi:providers:aws",
+					RetainOnDelete: true,
+					IgnoreChanges:  []string{"fooIgnore"},
+					DeletedWith:    "123",
+					URN:            "urn:pulumi:stack::project::pulumi:providers:aws::default_123",
 				},
 				{
-					ID:     "123",
-					Custom: true,
-					Type:   "pulumi:providers:random",
-					URN:    "urn:pulumi:stack::project::pulumi:providers:random::default_123",
+					ID:             "123",
+					ImportID:       "abc",
+					Custom:         true,
+					Type:           "pulumi:providers:random",
+					RetainOnDelete: true,
+					IgnoreChanges:  []string{"fooIgnore"},
+					DeletedWith:    "123",
+					URN:            "urn:pulumi:stack::project::pulumi:providers:random::default_123",
 				},
 				{
-					ID:     "id",
-					Custom: true,
-					Type:   "pulumi:providers:pkg",
-					URN:    "urn:pulumi:stack::project::pulumi:providers:pkg::provider",
+					ID:             "id",
+					ImportID:       "abc",
+					Custom:         true,
+					Type:           "pulumi:providers:pkg",
+					RetainOnDelete: true,
+					IgnoreChanges:  []string{"fooIgnore"},
+					DeletedWith:    "123",
+					URN:            "urn:pulumi:stack::project::pulumi:providers:pkg::provider",
```

Comment:
I feel like we should add some resources that _don't_ set all the properties would be good too, just to check?